@noun-auction/components/AuctionUi/NounsBidFormComponent.tsx

```diff
@@ -0,0 +1,176 @@
+import { useIsAuctionCompleted } from '@noun-auction/hooks/useIsAuctionCompleted'
+import { Box, Flex, Grid, Input, Label, Separator, Stack } from '@zoralabs/zord'
+
+export type NounsBidFormComponentProps = {
+  activeAuction: TypeSafeNounsAuction
+  handleOnSubmit: any // FIXME: find ugly type for this
+  isError: boolean
+  isSuccess: boolean
+  isLoading: boolean
+  setBidAmount: (s: string) => void
+  bidAmount: string
+  onConfirmation?: any
+  layout: 'row' | 'historyOnly' | 'withHistory' | 'sideBarBid'
+  errorComponent?: JSX.Element | null | false
+  prepareError: Error | null
+}
+
+export function NounsBidFormComponent({
+  onConfirmation,
+  layout,
+  bidAmount,
+  setBidAmount,
+  activeAuction,
+  handleOnSubmit,
+  isError,
+  isLoading,
+  isSuccess,
+  errorComponent,
+  prepareError,
+  ...props
+}: NounsBidFormComponentProps) {
+  const { address } = useAccount()
+  const { requestClose } = useModal()
+  const { isEnded: auctionCompleted, countdownText } = useIsAuctionCompleted({
+    activeAuction,
+  })
+  const { minBidAmount } = useNounBidIncrement(
+    activeAuction.reservePrice,
+    activeAuction.highestBidPrice?.chainTokenPrice?.raw,
+    activeAuction.minBidIncrementPercentage
+  )
+
+  const handleOnUpdate = useCallback(
+    (value: string) => {
+      let newValue: EthersBN
+      try {
+        newValue = parseUnits(value, 18)
+        const bidString = newValue.toString()
+        setBidAmount(bidString)
+      } catch (e) {
+        console.error(e)
+        return
+      }
+    },
+    [setBidAmount]
+  )
+
+  const bid = EthersBN.from(bidAmount)
+  const min = EthersBN.from(minBidAmount.raw)
+  const isSufficientBid = bidAmount === '0' ? false : bid.gte(min)
+
+  return (
+    <Box {...props}>
+      <form onSubmit={handleOnSubmit}>
+        <Flex justify="space-between">
+          <Label color="text2" size="lg">
+            Bid
+          </Label>
+        </Flex>
+        <Flex py="x2" mb="x4">
+          <Input
+            type="text"
+            min={minBidAmount?.pretty}
+            disabled={isSuccess}
+            pattern="[0-9.]*"
+            placeholder={`${minBidAmount?.pretty} Ξ or more`}
+            sizeVariant="lg"
+            onChange={(event: any) => handleOnUpdate(event.target.value)}
+          />
+        </Flex>
+        <Stack gap="x4" mb="x4">
+          <AuctionCountdown
+            auctionCompleted={auctionCompleted}
+            auctionEndTime={activeAuction.endTime}
+            layoutDirection="row"
+            layout="row"
+            showLabels
+            styles={{ justify: 'space-between' }}
+          />
+          <Separator />
+          <AuctionHighBid
+            auctionCompleted={auctionCompleted}
+            highestBid={activeAuction.highestBidPrice?.nativePrice?.raw}
+            collectionAddress={activeAuction.collectionAddress}
+            layout={layout}
+            showLabels
+            styles={{
+              layoutDirection: 'row',
```

Comment:
Again, layoutDirection is not a CSS property

If we're using styles={{}}..., we should look at whether we can change the component's props so it extends from Flex, Stack, or BoxProps.

eg:

```
type AuctionBidderProps = {
  label?: string
  layoutDirection?: 'row' | 'column'
  showLabels?: boolean
  useAvatar?: boolean
  layout: keyof typeof auctionWrapperVariants['layout']
  highestBidder?: string
  className?: string
  styles: {
    [key: string]: any
  }
}
```

we could do:

```
interface AuctionBidderProps extends FlexProps {
  label?: string
  showLabels?: boolean
  useAvatar?: boolean
  layout: keyof typeof auctionWrapperVariants['layout']
  highestBidder?: string
  className?: string
  styles: {
    [key: string]: any
  }
}
```

Reply:
(and same with justify="space-between")

Just cleaner and more consistent with the rest of the project